### PR TITLE
Allow to close opened diagram tabs

### PR DESCRIPTION
**Pull request by SteveMonnier**
A blue cross icon has been added on the diagram tab to be able to close
them.
RAF: Closing the last opened diagram automatically reopens it because of
the representationId kept by the method useParams in EditProjectView.
Also, there is some extra space after the cross icon that should be
removed.

Change-Id: I4d2b8532b4df312c4526cc91d1a9fe525f3d6860
Signed-off-by: Steve Monnier <steve.monnier@obeo.fr>

### CODE REVIEW

File: frontend/src/views/edit-project/EditProjectView.tsx [hunk 1]
```diff
@@ -109,6 +110,11 @@ export const EditProjectView = () => {
     if (displayedRepresentation && displayedRepresentation.id !== representationId) {
       const pathname = generatePath(routeMatch.path, { projectId, representationId: displayedRepresentation.id });
       history.push({ pathname });
+    } else if (displayedRepresentation === undefined) {
```
Comment (sbegaudeau) [inline frontend/src/views/edit-project/EditProjectView.tsx:113]: `if (!displayedRepresentation) {` in order to both consider `null` and `undefined`

File: frontend/src/views/edit-project/RepresentationNavigation.module.css [hunk 2]
```diff
@@ -17,6 +17,7 @@
 }
 
 .item {
+  columns: 2;
```
Comment (sbegaudeau) [inline frontend/src/views/edit-project/RepresentationNavigation.module.css:20]: Do not use CSS `column` layout, start with Flexbox and if really necessary to Grid

File: frontend/src/views/edit-project/__tests__/reducer.test.ts [hunk 3]
```diff
@@ -295,6 +309,38 @@ describe('EditProjectView - reducer', () => {
...
+      viewState: PROJECT_LOADED__STATE,
+      project: projectLoadedAndrepresentationDisplayedState.project,
+      representations: [],
+      displayedRepresentation: undefined,
+      selection: undefined,
+      subscribers: [],
+      message: undefined,
+    });
+  });
+
```
Comment (sbegaudeau) [inline frontend/src/views/edit-project/__tests__/reducer.test.ts:343]: Could you add a test for the last possibility of your `if / else` in `handleRepresentationClosedAction`?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
- You do not need to use a changeId, we are not using Gerrit anymore
- Please reference a Github issue in your commit message

**Review by sbegaudeau — APPROVED** [review]

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
Sorry, wrong button -.-

File: frontend/src/views/edit-project/EditProjectView.tsx (hunk 1 above)
Comment (SteveMonnier) [reply frontend/src/views/edit-project/EditProjectView.tsx]: Done

File: frontend/src/views/edit-project/RepresentationNavigation.module.css (hunk 2 above)
Comment (SteveMonnier) [reply frontend/src/views/edit-project/RepresentationNavigation.module.css]: Done

File: frontend/src/views/edit-project/__tests__/reducer.test.ts (hunk 3 above)
Comment (SteveMonnier) [reply frontend/src/views/edit-project/__tests__/reducer.test.ts:372]: Done

**Comment by SteveMonnier** [comment]
> * o

**Comment by SteveMonnier** [comment]
I am not sure how to reply to one of your previous comment directly, but I removed the  changeId I added by reflex and added the reference to a Github issue I just created.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
The second commit regarding edge edition does not belong in this pull request

File: frontend/src/views/edit-project/EditProjectView.tsx [hunk 4]
```diff
@@ -109,6 +110,12 @@ export const EditProjectView = () => {
     if (displayedRepresentation && displayedRepresentation.id !== representationId) {
       const pathname = generatePath(routeMatch.path, { projectId, representationId: displayedRepresentation.id });
       history.push({ pathname });
+    } else if (!displayedRepresentation && performance.navigation.type !== PerformanceNavigation.TYPE_RELOAD) {
```
Comment (pcdavid) [inline frontend/src/views/edit-project/EditProjectView.tsx:113]: We should not have to depend on the "performance" API here. I guess the issue (and fix) is similar to what you did for the OnboardArea bug, i.e. refine the state machine.

File: frontend/src/views/edit-project/EditProjectView.tsx [hunk 5]
```diff
@@ -109,6 +110,12 @@ export const EditProjectView = () => {
...
+      //If the browser was reloaded there is no need to generate a new path
```
Comment (pcdavid) [inline frontend/src/views/edit-project/EditProjectView.tsx:114]: The issue is not about reloading.

**Review by pcdavid — CHANGES_REQUESTED** [review]

**Comment by sbegaudeau** [comment]
Will be merged as part of https://github.com/eclipse-sirius/sirius-components/pull/294

**Pull request closed without merging**